src/basic/DAXPY.hpp

```diff
@@ -20,11 +20,53 @@
 #define DAXPY_DATA_SETUP \
   Real_ptr x = m_x; \
   Real_ptr y = m_y; \
-  Real_type a = m_a;
+  const Real_type a = m_a;
 
 #define DAXPY_BODY  \
   y[i] += a * x[i] ;
 
+#define DAXPY_DATA_VEC_SETUP \
+  RAJA_INDEX_VALUE_T(I, Int_type, "I");\
+  using vector_t = RAJA::StreamVector<Real_type, 2>; \
+  RAJA::TypedView<Real_type, RAJA::Layout<1, Int_type, 0>, I> X(x, iend); \
+  RAJA::TypedView<Real_type, RAJA::Layout<1, Int_type, 0>, I> Y(y, iend);
+
+#define DAXPY_DATA_VEC_SETUP2 \
+  RAJA_INDEX_VALUE_T(I, Int_type, "I"); \
+  using vector_t = RAJA::StreamVector<Real_type,2>; \
+  RAJA::TypedView<Real_type, RAJA::Layout<1, Int_type, 0>, I> Xview(x, iend); \
+  RAJA::TypedView<Real_type, RAJA::Layout<1, Int_type, 0>, I> Yview(y, iend); \
+  RAJA::forall<RAJA::vector_exec<vector_t>> (RAJA::TypedRangeSegment<I>(ibegin, iend),\
+  [=](RAJA::VectorIndex<I, vector_t> i) { \
+    vector_t X(0), Y(0); \
+    for(int j = 0; j < i.size(); ++j) { \
+      X.set(j, *(x + (**i) + j)); \
+      Y.set(j, *(y + (**i) + j)); \
+    } \
+    Xview(i) = X; \
+    Yview(i) = Y; \
+  });
+
+#define DAXPY_DATA_VEC_SETUP3 \
+  RAJA_INDEX_VALUE_T(I, Int_type, "I");\
+  using element_t = RAJA::StreamVector<Real_type,2>::element_type; \
+  element_t X[iend], Y[iend]; \
```

Comment:
This really isn't the intended use.  You are basically telling the compiler to hold the entire array X and Y in register at the same time.
You want to keep they arrays as "Real_type X[iend], Y[iend]", and load vector-sized chunks of those arrays using element_t.  You can do the load/stores either with Views+VectorIndexs, or by using the load()  and store() functions in the vector or register classes.